xrpl/utils/tx_parser/balance_changes.py

```diff
@@ -0,0 +1,231 @@
+                    field 'Destination' or is not included.'
+
+    Args:
+        metadata (`Dict[str, Union[str, int, bool, Dict[str, Any]]]`):
+            Transaction metadata including the account that
+            sent the transaction and the affected nodes.
+    """
+
+    def __init__(
+        self: Any, metadata: Dict[str, Union[str, int, bool, Dict[str, Any]]]
+    ) -> None:
+        """
+        Args:
+            metadata (`Dict[str, Union[str, int, bool, Dict[str, Any]]]`):
+                Transaction metadata including the account that
+                sent the transaction and the affected nodes.
+        """
+        super().__init__(metadata)
+
+        self._parse()
+
+    def _parse(self: Any) -> None:
+        is_valid_metadata(metadata=self.metadata)
+        parsedQuantities = parse_quantities(
+            metadata=self.metadata, valueParser=compute_balance_changes
+        )
+
+        result: Dict[str, List[Dict[str, str]]] = {}
+        for k, v in parsedQuantities.items():
+            address = k
+            change = v
+            result[address] = []
+            for obj in change:
+                if isinstance(obj.counterparty, tuple):
+                    obj.counterparty = obj.counterparty[0]
+                result[address].append(
+                    {
+                        "Counterparty": obj.counterparty,
+                        "Currency": obj.currency,
+                        "Value": obj.value,
+                    }
+                )
+
+        self._define_results(result=result)
+
+
+class ParseFinalBalances(BalanceChanges):
+    """Parse the final balances of all accounts affected
+    by the transaction after it occurred.
+
+    Usage:
+        Parse all final balances:
+            ``ParseFinalBalances(metadata=tx).all_balances``
+        Parse final balances of the account that sent the transaction:
+            ``ParseFinalBalances(metadata=tx).source_account_balances``
+        Parse final balances of the account that received the transaction:
+            ``ParseFinalBalances(metadata=tx).destination_account_balances``
+                If provided transaction has no field `Destination`:
+                    'Transaction type: `{tx_type}` has no
+                    field 'Destination' or is not included.'
+
+    Attributes:
+        metadata (`Dict[str, Union[str, int, bool, Dict[str, Any]]]`):
+            Transaction metadata including the account that
+            sent the transaction and the affected nodes.
+    """
+
+    def __init__(
+        self: Any, metadata: Dict[str, Union[str, int, bool, Dict[str, Any]]]
+    ) -> None:
+        """
+        Attributes:
+            metadata (`Dict[str, Union[str, int, bool, Dict[str, Any]]]`):
+                Transaction metadata including the account that
+                sent the transaction and the affected nodes.
+        """
+        super().__init__(metadata)
+
+        self._parse()
+
+    def _parse(self: Any) -> None:
```

Comment:
If you'd like to have `self: ParseFinalBalances` instead of `Any`, you just need to put `import __future__` at the top of the file. That allows python to see that you defined `ParseFinalBalances` even if at this point in the file the definition isn't done yet.

Reply:
Ah, I wondered how I would do that 👍🏼